thorlcr/activities/hashdistrib/thhashdistribslave.cpp

```diff
@@ -2924,12 +3060,29 @@ class HashDedupSlaveActivityBase : public CSlaveActivity
                 lastEog = false;
                 if (bucketHandler->addRow(row)) // true if new, i.e. non-duplicate (does not take ownership)
                 {
-                    dataLinkIncrement();
-                    return row.getClear();
+                    if (!keepBest)
+                    {
+                        dataLinkIncrement();
+                        return row.getClear();
+                    }
                 }
             }
             else
             {
+                if (keepBest && !keepBestRowsReadyToReturn)
+                {
+                    bucketHandler->clearCallbacks();
+                    bestRowStream.setown(bucketHandler->getRowStream());
+                    const void * row = bestRowStream->nextRow();
+                    if (row)
+                    {
+                        keepBestRowsReadyToReturn = true;
+                        LinkThorRow(row);
```

Comment:
IRowStream::nextRow() should always return a linked row.

I think your stream does - so I think you're leaking here by linking again.